File: src/main/java/com/glitch/knowledge/rest/HomeController.java
package com.glitch.knowledge.rest;

import java.util.List;
import java.util.Locale;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import com.glitch.knowledge.model.Station;
import com.glitch.knowledge.service.StationService;
import com.glitch.knowledge.util.LanguageUtil;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

@Controller
public class HomeController {

    @Autowired
    private StationService stationService;

    @Value("${application.use-mins}")
    private boolean useMins;

    @GetMapping("/")
    public String knowledge(@RequestParam(required = false) String lang, HttpServletRequest request,
            HttpServletResponse response, Locale locale, Model model) {
        List<Station> stations = stationService.getTwoRandom();

        String final_lang = LanguageUtil.getLanguageAndSetCookie(lang, locale, request, response);

        model.addAttribute("origin", stations.get(0));
        model.addAttribute("destination", stations.get(1));
        model.addAttribute("lang", final_lang);
        model.addAttribute("useMins", this.useMins);

        return "index";
    }

}

File: src/main/kotlin/com/glitch/knowledge/rest/HomeController.kt
package com.glitch.knowledge.rest;

import com.glitch.knowledge.model.Station
import com.glitch.knowledge.service.StationService
import com.glitch.knowledge.util.getLanguageAndSetCookie
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import java.util.*

@Controller
class HomeController(
    private val stationService: StationService,
    @param:Value("\${application.use-mins}") private val useMins: Boolean = false
) {

    @GetMapping("/")
    fun knowledge(
        @RequestParam(required = false) lang: String?,
        request: HttpServletRequest,
        response: HttpServletResponse,
        locale: Locale,
        model: Model
    ): String {
        val stationData: List<Station> = this.stationService.getTwoRandom()

        val finalLang: String = getLanguageAndSetCookie(lang, locale, request, response)

        model.addAttribute("origin", stationData[0])
        model.addAttribute("destination", stationData[1])
        model.addAttribute("lang", finalLang)
        model.addAttribute("useMins", this.useMins)

        return "index"
    }

}